src/pages/proposals/[id].tsx

```diff
@@ -471,59 +479,83 @@ interface CalldataRowsData {
   calldatasParsed: DecodedData[]
 }
 
-const CalldataRows = ({ calldatasParsed }: CalldataRowsData) => {
-  return calldatasParsed.map((callData, index) => <CalldataDisplay key={index} {...callData} />)
-}
-
-const CalldataDisplay = ({ functionName, args, inputs }: DecodedData) => (
-  <div>
-    <span className="flex justify-between">
-      <Paragraph variant="semibold" className="text-[16px] text-left">
-        Function:
-      </Paragraph>
-      <Span className="font-normal text-left">{functionName}</Span>
-    </span>
-
-    <Paragraph variant="semibold" className="text-[16px] mt-2">
-      Arguments:
-    </Paragraph>
-    <ul>
-      {inputs.map((input, index) => {
-        const inputName = input.name
-        const functionInputNames =
-          actionInputNameFormatMap[functionName] ||
-          ({} as InputNameFormatMap<typeof functionName, typeof inputName>)
-        const formattedInputName = (functionInputNames[inputName as never] || inputName) as string
-
-        const inputValue = args[index] as InputParameterTypeByFnByName<typeof functionName, typeof inputName>
-        const inputValueComposerMap = (actionComponentMap[functionName] || {}) as InputValueComposerMap<
-          typeof functionName,
-          typeof inputName
-        >
-        const InputComponent = inputValueComposerMap[
-          inputName as keyof typeof inputValueComposerMap
-        ] as InputValueComponent<InputParameterTypeByFnByName<typeof functionName, typeof inputName>>
-
-        return (
-          <li key={index} className="my-2 flex justify-between">
-            <Typography tagVariant="span" className="font-semibold text-[16px] text-left">
-              {formattedInputName}
-            </Typography>
-            {InputComponent && (
-              <InputComponent
-                value={inputValue}
-                htmlProps={{
-                  className: 'font-normal text-right',
-                }}
-              />
-            )}
-          </li>
-        )
-      })}
-    </ul>
-  </div>
+const CalldataRows = React.memo(
+  ({ calldatasParsed }: CalldataRowsData) => {
+    return (
+      <>
+        {calldatasParsed.map((callData, index) => (
+          <CalldataDisplay key={index} {...callData} />
+        ))}
+      </>
+    )
+  },
+  (prevProps, nextProps) => {
+    return prevProps.calldatasParsed === nextProps.calldatasParsed
+  },
 )
 
+CalldataRows.displayName = 'CalldataRows' // For debugging purposes
+
+const CalldataDisplay = ({
+  functionName,
+  args,
+  inputs,
+  affectedAddress,
+  callData,
+}: DecodedData & { affectedAddress?: string; callData?: string }) => {
```

Comment:
Can you create a type for this affectedAddress; callData

And then use it here
and in [id].tsx -> CalldataDisplay:505